src/edu/csus/ecs/pc2/core/Utilities.java

```diff
@@ -1903,8 +1903,61 @@ public static long convertCLICSContestTimeToMS(String time) {
         }
 
         return list;
-    }
 
-    
+    }    
   
+    /**
+     * Accepts a "duration" (an amount of time) in milliseconds and returns that duration in formatted form.
+     * The general form of the returned format is HHHH:MM:SS.sss, where the number of hour digits will always be
+     * at least two, but may be more depending on the input value 
+     * (in other words, this is not a "time of day" which is restricted to just two hour-digits).
+     * Note however that the minutes and seconds fields will always be two digits, and the msec field will 
+     * always be three digits.
+     * 
+     * @param milliseconds the duration to be formatted, in msec.
+     * @return the formatted value of the input duration.
+     */
+    public static String formatDuration (long milliseconds) {
+        
+        String result = "";
+        
+        long msecPerSecond = 1000;
```

Comment:
Consider changing to time constants in Constants class, ex. Constants.MS_PER_SECOND, Constants.MS_PER_MINUTE